appx.c

```diff
@@ -0,0 +1,2645 @@
+static const u_char APPXBUNDLE_UUID[] = { 0xB3, 0x58, 0x5F, 0x0F, 0xDE, 0xAA, 0x9A, 0x4B, 0xA4, 0x34, 0x95, 0x74, 0x2D, 0x92, 0xEC, 0xEB, };
+
+static const char PKCX_SIGNATURE[4] = { 'P', 'K', 'C', 'X' }; //Main header header
+static const char APPX_SIGNATURE[4] = { 'A', 'P', 'P', 'X' }; //APPX header
+static const char AXPC_SIGNATURE[4] = { 'A', 'X', 'P', 'C' }; //digest of zip file records
+static const char AXCD_SIGNATURE[4] = { 'A', 'X', 'C', 'D' }; //digest zip file central directory
+static const char AXCT_SIGNATURE[4] = { 'A', 'X', 'C', 'T' }; //digest of uncompressed [ContentTypes].xml
+static const char AXBM_SIGNATURE[4] = { 'A', 'X', 'B', 'M' }; //digest of uncompressed AppxBlockMap.xml
+static const char AXCI_SIGNATURE[4] = { 'A', 'X', 'C', 'I' }; //digest of uncompressed AppxMetadata/CodeIntegrity.cat (optional)
+
+static const char *HASH_METHOD_TAG = "HashMethod";
+static const char *HASH_METHOD_SHA256 = "http://www.w3.org/2001/04/xmlenc#sha256";
+static const char *HASH_METHOD_SHA384 = "http://www.w3.org/2001/04/xmldsig-more#sha384";
+static const char *HASH_METHOD_SHA512 = "http://www.w3.org/2001/04/xmlenc#sha512";
+
+#define EOCDR_SIZE 22
+#define ZIP64_EOCD_LOCATOR_SIZE 20
+#define ZIP64_HEADER 0x01
+#define COMPRESSION_NONE 0
+#define COMPRESSION_DEFLATE 8
+#define DATA_DESCRIPTOR_BIT (1 << 3)
+
+typedef struct zipLocalHeader_s
+{
+	uint16_t version;
+	uint16_t flags;
+	uint16_t compression;
+	uint16_t modTime;
+	uint16_t modDate;
+	uint32_t crc32;
+	uint64_t compressedSize;
+	uint64_t uncompressedSize;
+	uint16_t fileNameLen;
+	uint16_t extraFieldLen;
+	char *fileName;
+	uint8_t *extraField;
+
+	bool compressedSizeInZip64;
+	bool uncompressedSizeInZip64;
+} zipLocalHeader_t;
+
+typedef struct zipOverrideData_s
+{
+	uint32_t crc32;
+	uint64_t compressedSize;
+	uint64_t uncompressedSize;
+	uint8_t *data;
+} zipOverrideData_t;
+
+typedef struct zipCentralDirectoryEntry_s
+{
+	uint16_t creatorVersion;
+	uint16_t viewerVersion;
+	uint16_t flags;
+	uint16_t compression;
+	uint16_t modTime;
+	uint16_t modDate;
+	uint32_t crc32;
+	uint64_t compressedSize;
+	uint64_t uncompressedSize;
+	uint16_t fileNameLen;
+	uint16_t extraFieldLen;
+	uint16_t fileCommentLen;
+	uint32_t diskNoStart;
+	uint16_t internalAttr;
+	uint32_t externalAttr;
+	uint64_t offsetOfLocalHeader;
+	char *fileName;
+	uint8_t *extraField;
+	char *fileComment;
+	int64_t fileOffset;
+	int64_t entryLen;
+
+	bool compressedSizeInZip64;
+	bool uncompressedSizeInZip64;
+	bool offsetInZip64;
+	bool diskNoInZip64;
+
+	zipOverrideData_t *overrideData;
+
+	struct zipCentralDirectoryEntry_s *next;
+} zipCentralDirectoryEntry_t;
+
+typedef struct zip64EOCDR_s
+{
+	uint64_t eocdrSize;
+	uint16_t creatorVersion;
+	uint16_t viewerVersion;
+	uint32_t diskNumber;
+	uint32_t diskWithCentralDirectory;
+	uint64_t diskEntries;
+	uint64_t totalEntries;
+	uint64_t centralDirectorySize;
+	uint64_t centralDirectoryOffset;
+	int64_t commentLen;
+	char *comment;

```

Comment:
This field is named "zip64 extensible data sector" in the zip specification, it's more an opaque byte array than a string comment.